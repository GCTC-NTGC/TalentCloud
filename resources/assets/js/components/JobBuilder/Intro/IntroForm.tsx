import React, { useState } from "react";
import {
  injectIntl,
  WrappedComponentProps,
  FormattedMessage,
  defineMessages,
} from "react-intl";
import { Form, Field, Formik } from "formik";
import * as Yup from "yup";
import nprogress from "nprogress";
import get from "lodash/get";
import { validationMessages } from "../../Form/Messages";
import { Job, Department, Manager } from "../../../models/types";
import { emptyJob } from "../../../models/jobUtil";
import SelectInput from "../../Form/SelectInput";
import TextInput from "../../Form/TextInput";
import { getId } from "../../../helpers/queries";

const pageMessages = defineMessages({
  explanationBoldText: {
    id: "jobBuilder.intro.explanation.boldText",
    defaultMessage: "confirm that your personal information below is correct.",
    description: "Bold text portion of the JPB Intro explanation.",
  },
  emailLinkTitle: {
    id: "jobBuilder.intro.emailLinkTitle",
    defaultMessage: "Email Talent Cloud",
    description: "Title of the Talent Cloud email link.",
  },
  emailLinkText: {
    id: "jobBuilder.intro.emailLinkText",
    defaultMessage: "Talent Cloud",
    description: "Visible text of the Talent Cloud email link.",
  },
});

const formMessages = defineMessages({
  jobTitleLabelEN: {
    id: "jobBuilder.intro.jobTitleLabelEN",
    defaultMessage: "{name}'s Position (English)",
    description: "The label displayed on the manager position input.",
  },
  jobTitlePlaceholderEN: {
    id: "jobBuilder.intro.jobTitlePlaceholderEN",
    defaultMessage: "e.g. Design Manager",
    description: "The placeholder displayed on the Manager Position input.",
  },
  jobTitleLabelFR: {
    id: "jobBuilder.intro.jobTitleLabelFR",
    defaultMessage: "{name}'s Position (Français)",
    description: "The label displayed on the Manager Position input.",
  },
  jobTitlePlaceholderFR: {
    id: "jobBuilder.intro.jobTitlePlaceholderFR",
    defaultMessage: "e.g. Gestionnaire de la conception",
    description: "The placeholder displayed on the Manager Position input.",
  },
  departmentLabel: {
    id: "jobBuilder.intro.departmentLabel",
    defaultMessage: "{name}'s Department",
    description: "The label displayed on the department select box.",
  },
  departmentNullSelection: {
    id: "jobBuilder.intro.departmentNullSelection",
    defaultMessage: "Select a department...",
    description:
      "The default selection option displayed on the department select box.",
  },
  divisionLabelEN: {
    id: "jobBuilder.intro.divisionLabelEN",
    defaultMessage: "{name}'s Division (English)",
    description: "The label displayed on the division input.",
  },
  divisionPlaceholderEN: {
    id: "jobBuilder.intro.divisionPlaceholderEN",
    defaultMessage: "e.g. Digital Change",
    description: "The placeholder displayed on the division input.",
  },
  divisionLabelFR: {
    id: "jobBuilder.intro.divisionLabelFR",
    defaultMessage: "{name}'s Division (Français)",
    description: "The label displayed on the division input.",
  },
  divisionPlaceholderFR: {
    id: "jobBuilder.intro.divisionPlaceholderFR",
    defaultMessage: "e.g. Changement numérique",
    description: "The placeholder displayed on the division input.",
  },
});

// shape of values used in Form
interface IntroFormValues {
  managerPositionEn: string;
  managerPositionFr: string;
  department: number | "";
  divisionEN: string;
  divisionFR: string;
}

interface IntroFormProps {
  // If not null, used to prepopulate form values
  job: Job | null;
  // The manager of this job. Used to prepopulate values.
  manager: Manager;
  // List of known department options.
  departments: Department[];
  // Runs after successful validation.
  // It must (asynchronously) return the resulting job, if successful.
  handleSubmit: (job: Job, manager: Manager) => Promise<Job>;
  // Continues to next step in JobBuilder.
  handleContinue: (chosenLang: "en" | "fr", job: Job) => void;
  // // Continues the JobBuilder in French.
  // handleContinueFr: (job: Job) => void;
}

const initializeValues = (
  job: Job | null,
  manager: Manager,
): IntroFormValues => {
  let department: number | "" = "";
  if (job !== null && job.department_id !== null) {
    department = job.department_id;
  } else if (manager.department_id !== null) {
    department = manager.department_id;
  }

  const managerDivision = {
    en: get(manager, "division.en", ""),
    fr: get(manager, "division.fr", ""),
  };

  let divisionEN = "";
  if (job !== null && job.division.en) {
    divisionEN = job.division.en;
  } else if (managerDivision.en) {
    divisionEN = managerDivision.en;
  }

  let divisionFR = "";
  if (job !== null && job.division.fr) {
    divisionFR = job.division.fr;
  } else if (managerDivision.fr) {
    divisionFR = managerDivision.fr;
  }

  return {
    managerPositionEn: get(manager, "position.en", ""),
    managerPositionFr: get(manager, "position.fr", ""),
    department,
    divisionEN,
    divisionFR,
  };
};

const updateJobWithValues = (
  job: Job,
  values: IntroFormValues,
  locale: string,
): Job => ({
  ...job,
  // eslint-disable-next-line @typescript-eslint/camelcase
  chosen_lang: locale,
  // eslint-disable-next-line @typescript-eslint/camelcase
  department_id: values.department || null,
  division: {
    ...job.division,
    en: values.divisionEN || null,
    fr: values.divisionFR || null,
  },
});

const updateManagerWithValues = (
  manager: Manager,
  values: IntroFormValues,
): Manager => ({
  ...manager,
  // eslint-disable-next-line @typescript-eslint/camelcase
  department_id: values.department || null,
  division: {
    ...manager.division,
    en: values.divisionEN || null,
    fr: values.divisionFR || null,
  },
  position: {
    ...manager.position,
    en: values.managerPositionEn || null,
    fr: values.managerPositionFr || null,
  },
});

const IntroForm: React.FunctionComponent<IntroFormProps &
  WrappedComponentProps> = ({
  job,
  manager,
  departments,
  handleSubmit,
  handleContinue,
  intl,
}: IntroFormProps & WrappedComponentProps): React.ReactElement => {
  const { locale } = intl;
  if (locale !== "en" && locale !== "fr") {
    throw Error("Unexpected intl.locale"); // TODO: Deal with this more elegantly.
  }
  const initialValues: IntroFormValues = initializeValues(job, manager);
  const [languageSelection, setLanguageSelection] = useState(locale);

  const introSchema = Yup.object().shape({
    managerPositionEn: Yup.string().required(
      intl.formatMessage(validationMessages.required),
    ),
    managerPositionFr: Yup.string().required(
      intl.formatMessage(validationMessages.required),
    ),
    divisionEN: Yup.string().required(
      intl.formatMessage(validationMessages.required),
    ),
    divisionFR: Yup.string().required(
      intl.formatMessage(validationMessages.required),
    ),
    department: Yup.number()
      .oneOf(
        departments.map(getId),
        intl.formatMessage(validationMessages.invalidSelection),
      )
      .required(intl.formatMessage(validationMessages.required)),
  });

  return (
    <>
      <div data-c-container="form" data-c-padding="top(triple) bottom(triple)">
        <h3
          data-c-font-size="h3"
          data-c-font-weight="bold"
          data-c-margin="bottom(normal)"
        >
          <FormattedMessage
            id="jobBuilder.intro.welcome"
            defaultMessage="Welcome to the Job Poster Builder"
            description="Header of Job Poster Builder Intro Step"
          />
        </h3>
        <p data-c-margin="bottom(double)">
          <FormattedMessage
            id="jobBuilder.intro.explanation"
            defaultMessage="This tool will help you create a job poster that attracts the right talent. Before we get started on your job poster, take some time to {boldText}"
            description="Explanation of Job Poster Builder Intro Step"
            values={{
              boldText: (
                <span data-c-font-weight="bold">
                  {intl.formatMessage(pageMessages.explanationBoldText)}
                </span>
              ),
            }}
          />
        </p>
        <h4 data-c-font-size="h4" data-c-margin="bottom(normal)">
          <FormattedMessage
            id="jobBuilder.intro.formTitle"
            defaultMessage="{name}'s Profile Information"
            description="The title of the profile information form."
            values={{
              name: manager.first_name,
            }}
          />
        </h4>
        <p data-c-margin="bottom(double)">
          <FormattedMessage
            id="jobBuilder.intro.formDescription"
            defaultMessage="This information is used on the Job Poster to help applicants learn more about who they'll be working with."
            description="Explanation of why the profile information is collected."
          />
        </p>
        <Formik
          enableReinitialize
          initialValues={initialValues}
          validationSchema={introSchema}
          onSubmit={(values, { setSubmitting }): void => {
            const updatedJob = updateJobWithValues(
              job || emptyJob(),
              values,
              languageSelection,
            );
            const updatedManager = updateManagerWithValues(manager, values);
            nprogress.start();
            handleSubmit(updatedJob, updatedManager)
              .then((newJob: Job): void => {
                if (languageSelection === "fr") {
                  handleContinue("fr", newJob);
                } else {
                  handleContinue("en", newJob);
                }
              })
              .finally((): void => {
                nprogress.done();
                setSubmitting(false); // Required by Formik to finish the submission cycle
              });
          }}
          render={({
            isSubmitting,
            values,
            submitForm,
          }): React.ReactElement => (
            <>
              <Form id="form" data-c-margin="bottom(normal)">
                <div data-c-grid="gutter">
                  <Field
                    type="text"
                    id="builder01ManagerPositionEn"
                    name="managerPositionEn"
                    label={intl.formatMessage(formMessages.jobTitleLabelEN, {
                      name: manager.first_name,
                    })}
                    placeholder={intl.formatMessage(
                      formMessages.jobTitlePlaceholderEN,
                    )}
                    required
                    grid="tl(1of2)"
                    component={TextInput}
                  />
                  <Field
                    type="text"
                    id="builder01ManagerPositionFr"
                    name="managerPositionFr"
                    label={intl.formatMessage(formMessages.jobTitleLabelFR, {
                      name: manager.first_name,
                    })}
                    placeholder={intl.formatMessage(
                      formMessages.jobTitlePlaceholderFR,
                    )}
                    required
                    grid="tl(1of2)"
                    component={TextInput}
                  />
                  <Field
                    name="department"
                    id="builder01ManagerDepartment"
                    label={intl.formatMessage(formMessages.departmentLabel, {
                      name: manager.first_name,
                    })}
                    grid="base(1of1)"
                    component={SelectInput}
                    required
                    selected={values.department}
                    nullSelection={intl.formatMessage(
                      formMessages.departmentNullSelection,
                    )}
                    options={departments.map((dept: Department): {
                      value: number;
                      label: string;
                    } => ({
                      value: dept.id,
                      label: dept.name[locale],
                    }))}
                  />
                  <Field
                    type="text"
                    id="builder01ManagerDivisionEN"
                    name="divisionEN"
                    label={intl.formatMessage(formMessages.divisionLabelEN, {
                      name: manager.first_name,
                    })}
                    placeholder={intl.formatMessage(
                      formMessages.divisionPlaceholderEN,
                    )}
                    required
                    grid="tl(1of2)"
                    component={TextInput}
                  />
                  <Field
                    type="text"
                    id="builder01ManagerDivisionFR"
                    name="divisionFR"
                    label={intl.formatMessage(formMessages.divisionLabelFR, {
                      name: manager.first_name,
                    })}
                    placeholder={intl.formatMessage(
                      formMessages.divisionPlaceholderFR,
                    )}
                    required
                    grid="tl(1of2)"
                    component={TextInput}
                  />
                </div>
              </Form>
              <p data-c-margin="bottom(double)">
                <FormattedMessage
                  id="jobBuilder.intro.completeInLanguage"
                  defaultMessage="Complete the job poster in the language of your choice. We will
                  handle translation."
                  description="Instructions at bottom of form on language choice for job poster builder."
                />{" "}
                <FormattedMessage
                  id="jobBuilder.intro.contactUs"
                  defaultMessage="We’ve also provided instructions and examples to help guide you through the process but if you still have questions, contact {link}."
                  description="Subtitle 2 of Job Poster Builder Intro Step"
                  values={{
                    link: (
                      <a
                        href="mailto:talent.cloud-nuage.de.talents@tbs-sct.gc.ca"
                        title={intl.formatMessage(pageMessages.emailLinkTitle)}
                      >
                        {intl.formatMessage(pageMessages.emailLinkText)}
                      </a>
                    ),
                  }}
                />
              </p>
              {/* This button continues the user in English. */}
              <button
                form="form"
                data-c-button="solid(c1)"
                data-c-radius="rounded"
                data-c-margin="right(normal) bottom(normal)"
                type="button"
                disabled={isSubmitting}
                onClick={(): void => {
                  /** FIXME:
                   * This is a race condition, since setLanguageSelection is asynchronous.
                   * I have to find a way to handle 2 submit buttons in formik without a race condition somewhere :(
                   * For now, the setState always happens faster than the validation check, so it works.
                   * See https://github.com/jaredpalmer/formik/issues/214
                   * -- Tristan
                   */
                  setLanguageSelection("en");
                  submitForm();
                }}
              >
                <FormattedMessage
                  id="jobBuilder.intro.continueButtonLabelEN"
                  defaultMessage="Continue in English"
                  description="Label displayed on submit button for continuation of job builder in english."
                />
              </button>
              {/* This button switches the user to French and continues the process in French.  */}
              <button
                form="form"
                data-c-button="solid(c1)"
                data-c-radius="rounded"
                type="button"
                disabled={isSubmitting}
                onClick={(): void => {
                  /** FIXME:
                   * This is a race condition, since setLanguageSelection is asynchronous.
                   * I have to find a way to handle 2 submit buttons in formik without a race condition somewhere :(
                   * For now, the setState always happens faster than the validation check, so it works.
                   * See https://github.com/jaredpalmer/formik/issues/214
                   * -- Tristan
                   */
                  setLanguageSelection("fr");
                  submitForm();
                }}
              >
                <FormattedMessage
                  id="jobBuilder.intro.continueButtonLabelFR"
                  defaultMessage="Continuer en Français"
                  description="Label displayed on submit button for continuation of job builder in french."
                />
              </button>
            </>
          )}
        />
      </div>
    </>
  );
};

export default injectIntl(IntroForm);
