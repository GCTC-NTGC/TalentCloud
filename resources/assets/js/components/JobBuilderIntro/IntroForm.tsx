import React, { useState } from "react";
import {
  injectIntl,
  InjectedIntlProps,
  FormattedMessage,
  FormattedHTMLMessage,
  defineMessages,
} from "react-intl";
import { Form, Field, Formik, FormikValues } from "formik";
import * as Yup from "yup";
import Input from "../Forms/Input";
import Select from "../Forms/Select";
import { validationMessages } from "../Form/Messages";
import { Job } from "../../models/types";
import { DepartmentId } from "../../models/lookupConstants";
import { departmentName } from "../../models/localizedConstants";
import { emptyJob } from "../../models/jobUtil";

const formMessages = defineMessages({
  jobTitleLabelEN: {
    id: "jobBuilder.intro.jobTitleLabelEN",
    defaultMessage: "My Job Title (English)",
    description: "The label displayed on the job title input.",
  },
  jobTitlePlaceholderEN: {
    id: "jobBuilder.intro.jobTitlePlaceholderEN",
    defaultMessage: "e.g. Design Manager",
    description: "The placeholder displayed on the job title input.",
  },
  jobTitleLabelFR: {
    id: "jobBuilder.intro.jobTitleLabelFR",
    defaultMessage: "My Job Title (Français)",
    description: "The label displayed on the job title input.",
  },
  jobTitlePlaceholderFR: {
    id: "jobBuilder.intro.jobTitlePlaceholderFR",
    defaultMessage: "e.g. Gestionnaire de la conception",
    description: "The placeholder displayed on the job title input.",
  },
  departmentLabel: {
    id: "jobBuilder.intro.departmentLabel",
    defaultMessage: "My Department",
    description: "The label displayed on the department select box.",
  },
  departmentNullSelection: {
    id: "jobBuilder.intro.departmentNullSelection",
    defaultMessage: "Select a department...",
    description:
      "The default selection option displayed on the department select box.",
  },
  divisionLabelEN: {
    id: "jobBuilder.intro.divisionLabelEN",
    defaultMessage: "My Division (English)",
    description: "The label displayed on the division input.",
  },
  divisionPlaceholderEN: {
    id: "jobBuilder.intro.divisionPlaceholderEN",
    defaultMessage: "e.g. Digital Change",
    description: "The placeholder displayed on the division input.",
  },
  divisionLabelFR: {
    id: "jobBuilder.intro.divisionLabelFR",
    defaultMessage: "My Division (Français)",
    description: "The label displayed on the division input.",
  },
  divisionPlaceholderFR: {
    id: "jobBuilder.intro.divisionPlaceholderFR",
    defaultMessage: "e.g. Changement numérique",
    description: "The placeholder displayed on the division input.",
  },
});

// shape of values used in Form
interface FormValues {
  jobTitleEN: string;
  jobTitleFR: string;
  department: number | "";
  divisionEN: string;
  divisionFR: string;
}

interface IntroFormProps {
  // If not null, used to prepopulate form values
  job: Job | null;
  // Runs after successful validation.
  // It must (asyncronously) return true if the submission was successful, false otherwise.
  handleSubmit: (job: Job) => Promise<boolean>;
  // Continues the JobBuilder in English.
  handleContinueEn: () => void;
  // Continues the JobBuilder in French.
  handleContinueFr: () => void;
}

const jobToValues = (job: Job | null): FormValues =>
  job
    ? {
        jobTitleEN: job.en.title || "",
        jobTitleFR: job.fr.title || "",
        department: job.department_id || "",
        divisionEN: job.en.division || "",
        divisionFR: job.fr.division || "",
      }
    : {
        jobTitleEN: "",
        jobTitleFR: "",
        department: "",
        divisionEN: "",
        divisionFR: "",
      };

const updateJobWithValues = (job: Job, values: FormValues): Job => ({
  ...job,
  // eslint-disable-next-line @typescript-eslint/camelcase
  department_id: values.department || null,
  en: {
    ...job.en,
    title: values.jobTitleEN || null,
    division: values.divisionEN || null,
  },
  fr: {
    ...job.fr,
    title: values.jobTitleFR || null,
    division: values.divisionFR || null,
  },
});

const IntroForm: React.FunctionComponent<
  IntroFormProps & InjectedIntlProps
> = ({
  job,
  handleSubmit,
  handleContinueEn,
  handleContinueFr,
  intl,
}: IntroFormProps & InjectedIntlProps): React.ReactElement => {
  const initialValues: FormValues = jobToValues(job);
  const [languageSelection, setLanguageSelection] = useState("en");

  const introSchema = Yup.object().shape({
    jobTitleEN: Yup.string().required(
      intl.formatMessage(validationMessages.required),
    ),
    jobTitleFR: Yup.string().required(
      intl.formatMessage(validationMessages.required),
    ),
    divisionEN: Yup.string().required(
      intl.formatMessage(validationMessages.required),
    ),
    divisionFR: Yup.string().required(
      intl.formatMessage(validationMessages.required),
    ),
    department: Yup.number()
      .oneOf(
        Object.values(DepartmentId),
        intl.formatMessage(validationMessages.invalidSelection),
      )
      .required(intl.formatMessage(validationMessages.required)),
  });

  return (
    <>
      <div data-c-container="form" data-c-padding="top(triple) bottom(triple)">
        <h3
          data-c-font-size="h3"
          data-c-font-weight="bold"
          data-c-margin="bottom(normal)"
        >
          <FormattedMessage
            id="jobBuilder.intro.welcome"
            defaultMessage="EN Welcome to the Job Poster Builder"
            description="Header of Job Poster Builder Intro Step"
          />
        </h3>
        <p data-c-margin="bottom(normal)">
          <FormattedMessage
            id="jobBuilder.intro.subtitle01"
            defaultMessage="This tool will help you create a job poster that attracts the right talent."
            description="Subtitle 1 of Job Poster Builder Intro Step"
          />
        </p>
        <p data-c-margin="bottom(double)">
          <FormattedMessage
            id="jobBuilder.intro.subtitle02"
            defaultMessage="We’ve also provided instructions and examples to help guide you through the process but if you still have questions, contact "
            description="Subtitle 2 of Job Poster Builder Intro Step"
          />
          <a
            href="mailto:talent.cloud-nuage.de.talents@tbs-sct.gc.ca"
            title="Email Talent Cloud."
          >
            Talent Cloud
          </a>
        </p>
        <h4
          data-c-colour="c3"
          data-c-font-size="h4"
          data-c-margin="bottom(double)"
        >
          <FormattedMessage
            id="jobBuilder.intro.beforeWeStart"
            defaultMessage="Before we get started please review your information."
            description="Before starting header of Job Poster Builder Intro Step"
          />
        </h4>
        <Formik
          initialValues={initialValues}
          validationSchema={introSchema}
          onSubmit={(values, { setSubmitting }): void => {
            handleSubmit(updateJobWithValues(job || emptyJob(), values))
              .then((isSuccessful: boolean): void => {
                if (isSuccessful) {
                  if (languageSelection === "fr") {
                    handleContinueFr();
                  } else {
                    handleContinueEn();
                  }
                }
              })
              .finally(
                (): void => setSubmitting(false), // Required by Formik to finish the submission cycle
              );
          }}
          render={({
            isSubmitting,
            values,
            submitForm,
          }): React.ReactElement => (
            <>
              <Form id="form" data-c-margin="bottom(normal)">
                <div data-c-grid="gutter">
                  <Field
                    type="text"
                    htmlId="builder01ManagerJobTitleEN"
                    name="jobTitleEN"
                    label={intl.formatMessage(formMessages.jobTitleLabelEN)}
                    placeholder={intl.formatMessage(
                      formMessages.jobTitlePlaceholderEN,
                    )}
                    required
                    grid="tl(1of2)"
                    component={Input}
                  />
                  <Field
                    type="text"
                    htmlId="builder01ManagerJobTitleFR"
                    name="jobTitleFR"
                    label={intl.formatMessage(formMessages.jobTitleLabelFR)}
                    placeholder={intl.formatMessage(
                      formMessages.jobTitlePlaceholderFR,
                    )}
                    required
                    grid="tl(1of2)"
                    component={Input}
                  />
                  <Field
                    name="department"
                    id="builder01ManagerDepartment"
                    label={intl.formatMessage(formMessages.departmentLabel)}
                    grid="base(1of1)"
                    component={Select}
                    required
                    selected={values.department}
                    nullSelection={intl.formatMessage(
                      formMessages.departmentNullSelection,
                    )}
                    options={Object.values(DepartmentId).map((id: number): {
                      value: number;
                      label: string;
                    } => ({
                      value: id,
                      label: intl.formatMessage(departmentName(id)),
                    }))}
                  />
                  <Field
                    type="text"
                    htmlId="builder01ManagerDivisionEN"
                    name="divisionEN"
                    label={intl.formatMessage(formMessages.divisionLabelEN)}
                    placeholder={intl.formatMessage(
                      formMessages.divisionPlaceholderEN,
                    )}
                    required
                    grid="tl(1of2)"
                    component={Input}
                  />
                  <Field
                    type="text"
                    htmlId="builder01ManagerDivisionFR"
                    name="divisionFR"
                    label={intl.formatMessage(formMessages.divisionLabelFR)}
                    placeholder={intl.formatMessage(
                      formMessages.divisionPlaceholderFR,
                    )}
                    required
                    grid="tl(1of2)"
                    component={Input}
                  />
                </div>
              </Form>
              <p data-c-margin="bottom(normal)">
                <FormattedHTMLMessage
                  id="jobBuilder.intro.note"
                  defaultMessage="<strong>Note:</strong> Changing your information on this form will also change these fields in your Profile."
                  description="Ending note for hiring managers."
                />
              </p>
              <p data-c-margin="bottom(double)">
                <FormattedMessage
                  id="jobBuilder.intro.completeInLanguage"
                  defaultMessage="Complete the job poster in the language of your choice. We will
                  handle translation."
                  description="Intstructions at bottom of form on language choice for job poster builder."
                />
              </p>
              {/* This button continues the user in English. */}
              <button
                form="form"
                data-c-button="solid(c1)"
                data-c-radius="rounded"
                type="button"
                disabled={isSubmitting}
                onClick={(): void => {
                  setLanguageSelection("en");
                  submitForm();
                }}
              >
                <FormattedMessage
                  id="jobBuilder.intro.continueButtonLabelEN"
                  defaultMessage="Continue in English"
                  description="Label displayed on submit button for continuation of job builder in english."
                />
              </button>
              {/* This button switches the user to French and continues the process in French.  */}
              <button
                form="form"
                data-c-button="solid(c1)"
                data-c-radius="rounded"
                type="button"
                disabled={isSubmitting}
                onClick={(): void => {
                  setLanguageSelection("fr");
                  submitForm();
                }}
              >
                <FormattedMessage
                  id="jobBuilder.intro.continueButtonLabelFR"
                  defaultMessage="Continuer en Français"
                  description="Label displayed on submit button for continuation of job builder in french."
                />
              </button>
            </>
          )}
        />
      </div>
    </>
  );
};

export default injectIntl(IntroForm);
