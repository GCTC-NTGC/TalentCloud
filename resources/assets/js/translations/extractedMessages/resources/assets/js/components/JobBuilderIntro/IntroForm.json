[
  {
    "id": "jobBuilder.intro.explanation.boldText",
    "description": "Bolded text portion of the JPB Intro explanation.",
    "defaultMessage": "confirm that your personal information below is correct."
  },
  {
    "id": "jobBuilder.intro.emailLinkTitle",
    "description": "Title of the Talent Cloud email link.",
    "defaultMessage": "Email Talent Cloud"
  },
  {
    "id": "jobBuilder.intro.emailLinkText",
    "description": "Visible text of the Talent Cloud email link.",
    "defaultMessage": "Talent Cloud"
  },
  {
    "id": "jobBuilder.intro.jobTitleLabelEN",
    "description": "The label displayed on the manager position input.",
    "defaultMessage": "{name}'s Position (English)"
  },
  {
    "id": "jobBuilder.intro.jobTitlePlaceholderEN",
    "description": "The placeholder displayed on the Manager Position input.",
    "defaultMessage": "e.g. Design Manager"
  },
  {
    "id": "jobBuilder.intro.jobTitleLabelFR",
    "description": "The label displayed on the MKanager Position input.",
    "defaultMessage": "{name}'s Position (Français)"
  },
  {
    "id": "jobBuilder.intro.jobTitlePlaceholderFR",
    "description": "The placeholder displayed on the Manager Position input.",
    "defaultMessage": "e.g. Gestionnaire de la conception"
  },
  {
    "id": "jobBuilder.intro.departmentLabel",
    "description": "The label displayed on the department select box.",
    "defaultMessage": "{name}'s Department"
  },
  {
    "id": "jobBuilder.intro.departmentNullSelection",
    "description": "The default selection option displayed on the department select box.",
    "defaultMessage": "Select a department..."
  },
  {
    "id": "jobBuilder.intro.divisionLabelEN",
    "description": "The label displayed on the division input.",
    "defaultMessage": "{name}'s Division (English)"
  },
  {
    "id": "jobBuilder.intro.divisionPlaceholderEN",
    "description": "The placeholder displayed on the division input.",
    "defaultMessage": "e.g. Digital Change"
  },
  {
    "id": "jobBuilder.intro.divisionLabelFR",
    "description": "The label displayed on the division input.",
    "defaultMessage": "{name}'s Division (Français)"
  },
  {
    "id": "jobBuilder.intro.divisionPlaceholderFR",
    "description": "The placeholder displayed on the division input.",
    "defaultMessage": "e.g. Changement numérique"
  },
  {
    "id": "jobBuilder.intro.welcome",
    "description": "Header of Job Poster Builder Intro Step",
    "defaultMessage": "Welcome to the Job Poster Builder"
  },
  {
    "id": "jobBuilder.intro.explanation",
    "description": "Explanation of Job Poster Builder Intro Step",
    "defaultMessage": "This tool will help you create a job poster that attracts the right talent. Before we get started on your job poster, take some time to {boldText}"
  },
  {
    "id": "jobBuilder.intro.formTitle",
    "description": "The title of the profile information form.",
    "defaultMessage": "{name}'s Profile Information"
  },
  {
    "id": "jobBuilder.intro.formDescription",
    "description": "Explanation of why the profile information is collected.",
    "defaultMessage": "This information is used on the Job Poster to help applicants learn more about who they'll be working with."
  },
  {
    "id": "jobBuilder.intro.completeInLanguage",
    "description": "Intstructions at bottom of form on language choice for job poster builder.",
    "defaultMessage": "Complete the job poster in the language of your choice. We will\r\n                  handle translation."
  },
  {
    "id": "jobBuilder.intro.contactUs",
    "description": "Subtitle 2 of Job Poster Builder Intro Step",
    "defaultMessage": "We’ve also provided instructions and examples to help guide you through the process but if you still have questions, contact {link}."
  },
  {
    "id": "jobBuilder.intro.continueButtonLabelEN",
    "description": "Label displayed on submit button for continuation of job builder in english.",
    "defaultMessage": "Continue in English"
  },
  {
    "id": "jobBuilder.intro.continueButtonLabelFR",
    "description": "Label displayed on submit button for continuation of job builder in french.",
    "defaultMessage": "Continuer en Français"
  }
]