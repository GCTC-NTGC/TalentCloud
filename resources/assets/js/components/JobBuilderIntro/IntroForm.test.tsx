import React from "react";
import { shallow } from "enzyme";
import IntroForm, { InnerIntroForm } from "./IntroForm";

describe("Step0Form", (): void => {
  test("should render Step0Form correctly", (): void => {
    const wrapper = shallow(<IntroForm />);
    expect(wrapper).toMatchSnapshot();
  });

  test("should update an input when it is changed", (): void => {
    const wrapper = shallow(<IntroForm />);
    // console.log(wrapper.find());
    wrapper
      .find(InnerIntroForm)
      .dive()
      .find("Field")
      .simulate("change", {
        persist: () => {},

        target: {
          name: "jobTitleEN",
          value: "Web developer",
        },
      });

    const newValue = wrapper
      .find(InnerIntroForm)
      .dive()
      .find("Field")
      .props().value;

    expect(newValue).toEqual("Web developer");
  });
});
