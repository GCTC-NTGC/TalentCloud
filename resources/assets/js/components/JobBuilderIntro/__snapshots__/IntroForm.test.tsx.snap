// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Step0Form should render Step0Form correctly 1`] = `
<Formik
  displayName="IntroForm"
  enableReinitialize={false}
  handleSubmit={[Function]}
  initialValues={
    Object {
      "department": "",
      "divisionEN": "",
      "divisionFR": "",
      "jobTitleEN": "",
      "jobTitleFR": "",
    }
  }
  isInitialValid={false}
  onSubmit={[Function]}
  render={[Function]}
  validateOnBlur={true}
  validateOnChange={true}
  validationSchema={[Function]}
/>
`;
